File: apps/android/app/src/main/java/com/roundsloyalty/app/data/models/PurchaseClaim.kt
package com.roundsloyalty.app.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class PurchaseClaim(
    val id: String,
    @SerialName("customer_id") val customerId: String,
    @SerialName("vendor_id") val vendorId: String,
    @SerialName("receipt_image_url") val receiptImageUrl: String? = null,
    @SerialName("purchase_amount") val purchaseAmount: Double? = null,
    @SerialName("purchase_date") val purchaseDate: String? = null,
    @SerialName("customer_note") val customerNote: String? = null,
    @SerialName("vendor_note") val vendorNote: String? = null,
    val status: String = "pending_review",
    @SerialName("created_at") val createdAt: String? = null,
)

File: apps/ios/RoundsLoyalty/Models/PurchaseClaim.swift
import Foundation

enum ClaimStatus: String, Codable {
    case pendingReview = "pending_review"
    case approved
    case rejected
    case needsMoreInfo = "needs_more_info"
    case expired
}

struct PurchaseClaim: Codable, Identifiable {
    let id: UUID
    let customerId: UUID
    let vendorId: UUID
    let receiptImageUrl: String?
    let purchaseAmount: Double?
    let purchaseDate: String?
    let customerNote: String?
    let vendorNote: String?
    let status: ClaimStatus
    let createdAt: Date?
    var vendor: Vendor?

    enum CodingKeys: String, CodingKey {
        case id
        case customerId = "customer_id"
        case vendorId = "vendor_id"
        case receiptImageUrl = "receipt_image_url"
        case purchaseAmount = "purchase_amount"
        case purchaseDate = "purchase_date"
        case customerNote = "customer_note"
        case vendorNote = "vendor_note"
        case status
        case createdAt = "created_at"
        case vendor = "vendors"
    }
}
